src/shared_utilities/models.jl

```diff
@@ -418,5 +427,11 @@ function initialize(model::AbstractModel{FT}) where {FT}
     Y = initialize_prognostic(model, coords)
     p = initialize_auxiliary(model, coords)
     p = add_drivers_to_cache(p, model, coords)
+    p = add_dfluxBCdY_to_cache(
```

Comment:
similar, would not need this because the methods for `boundary_vars` change for the model/bc combo

Reply:
note that the `boundary_var` interface is only used for models with PDEs currently, where have/want the capability to set flexible boundary conditions, and the different types of BCs might require different variables. The canopy model is solving a set of ODEs without the same flexibility. If we want to ever implicitly timestep those equations, we would need some extra machinery, possibly like what you sketched out here. but what we have should also work if we need to implicitly timestep the soil & account for dfluxBCdY in the integrated soil/canopy case, since the soil has a different BC type in that case.